vignettes/visits_periods.Rmd

```diff
@@ -94,51 +94,194 @@ derive_vars_joined(
-) %>%
+ex <- tribble(
+  ~STUDYID, ~USUBJID,  ~VISIT,    ~EXTRT,   ~EXSTDTC,
+  "xyz",    "1",       "Day 1",   "Drug X", "2022-01-02",
+  "xyz",    "1",       "Week 4",  "Drug X", "2022-02-05",
+  "xyz",    "1",       "Week 8",  "Drug X", "2022-03-01",
+  "xyz",    "1",       "Week 12", "Drug X", "2022-04-03",
+  "xyz",    "1",       "Week 16", "Drug Y", "2022-05-03",
+  "xyz",    "1",       "Week 20", "Drug Y", "2022-06-02",
+  "xyz",    "1",       "Week 24", "Drug Y", "2022-07-01",
+  "xyz",    "1",       "Week 28", "Drug Y", "2022-08-04",
+  "xyz",    "2",       "Day 1",   "Drug Y", "2023-10-20",
+  "xyz",    "2",       "Week 4",  "Drug Y", "2023-11-21",
+  "xyz",    "2",       "Week 8",  "Drug Y", "2023-12-19",
+  "xyz",    "2",       "Week 12", "Drug Y", "2024-01-19",
+  "xyz",    "2",       "Week 16", "Drug X", "2024-02-20",
+  "xyz",    "2",       "Week 20", "Drug X", "2024-03-17",
+  "xyz",    "2",       "Week 24", "Drug X", "2024-04-22",
+  "xyz",    "2",       "Week 28", "Drug X", "2024-05-21"
+)
+
+ex
+```
+
+Then to create a period reference dataset, code like this (or similar) would suffice:
+
+```{r echo=TRUE}
+period_ref <- ex %>%
+  # Select visits marking the start of each period
+  filter(VISIT %in% c("Day 1", "Week 16")) %>%
+  # Create APERIOD, APERSDT, TRTA based on SDTM counterparts
+  mutate(
+    APERIOD = case_when(
+      VISIT == "Day 1" ~ 1,
+      VISIT == "Week 16" ~ 2
+    ),
+    APERSDT = ymd(EXSTDTC),
+    TRTA = EXTRT
+  ) %>%
+  # Create APEREDT based on start date of next period
+  arrange(USUBJID, desc(APERSDT)) %>%
+  group_by(USUBJID) %>%
   mutate(
-    STUDYID = "xyz",
-    APHASEN = as.integer(APHASEN),
-    across(matches("PH[ES]DT"), ymd)
+    APEREDT = lag(APERSDT) - 1 # one day before start of next period
   ) %>%
-  select(STUDYID, everything())
+  # Tidy up
+  ungroup() %>%
+  arrange(USUBJID, APERSDT) %>%
+  select(-starts_with("EX"), -VISIT)
 
-phase_ref
+period_ref
 ```
 
-The admiral functions expect separate datasets for periods, subperiods,
-and phases. For periods the numeric variable `APERIOD` is expected, for
-subperiods the numeric variables `APERIOD` and `ASPER`, and for phases
-the numeric variable `APHASEN`.
+The workflow above populates the Period End Date `APEREDT` for all periods 
+except the last one. The value of the last period could then be populated 
+with the End of Study Date (`EOSDT`) from ADSL, to obtain:
+
+```{r echo=TRUE}
+adsl <- tribble(
+  ~STUDYID,  ~USUBJID,  ~EOSDT,
+  "xyz",     "1",       ymd("2022-09-10"),
+  "xyz",     "2",       ymd("2024-06-30")
+)
+
+period_ref <- period_ref %>%
+  left_join(adsl, by = c("STUDYID", "USUBJID")) %>%
+  mutate(APEREDT = case_when(
+    APERIOD == "1" ~ APEREDT,
+    APERIOD == "2" ~ EOSDT
+  )) %>%
+  select(-EOSDT)
+
+period_ref
+```
+
+A very similar process could be followed for the phase reference dataset 
+for the study phases, this time using Day 1 and Week 28 as reference 
+visits, to obtain:
```

Comment:
I'm not sure about this because I like the relative simplicity of the section. If we add yet another workflow variant I think it could confuse things. If anything I would add this in another section? What do you think?

Reply:
I gave that a go - please let me know what you think. I also then removed a section that was duplicated. I would be interested in your opinion as to whether the whole vignette reads well now?